geotrek/tourism/forms.py

```diff
@@ -101,3 +130,19 @@ def __init__(self, *args, **kwargs):
         # Since we use chosen() in trek_form.html, we don't need the default help text
         for f in ['themes', 'source']:
             self.fields[f].help_text = ''
+        participants_count = { p.category.pk: p.count for p in self.instance.participants.all() }
+        for category in TouristicEventParticipantCategory.objects.all():
+            field_id = 'participant_count_{}'.format(category.id)
+            self.fields[field_id] = TouristicEventParticipantCount._meta.get_field('count').formfield(required=False)
+            self.fields[field_id].label = category.label
+            self.fields[field_id].initial = participants_count.get(category.pk)
+            self.fieldslayout[0][1][1][0].append(field_id)
+
+    def _save_m2m(self):
+        super()._save_m2m()
+        for category in TouristicEventParticipantCategory.objects.all():
+            count = self.cleaned_data['participant_count_{}'.format(category.id)]
+            if count is not None:
+                TouristicEventParticipantCount.objects.update_or_create(event=self.instance, category=category, defaults={'count': count})
+            else:
+                TouristicEventParticipantCount.objects.filter(event=self.instance, category=category).delete()
```

Comment:
Please add tests for forms for these two scenarios
See other `test_forms.py` files